### ISSUE TITLE
How to optimize geometry of excited states by dmrg?

### ISSUE THREAD

**Issue by jamcas** [pyscf/pyscf#338]
I try to refer to this example(https://github.com/pyscf/pyscf/blob/master/examples/geomopt/12-excited_states.py), but failed::
```
from pyscf import gto
from pyscf import scf, mcscf, dmrgscf
from pyscf import geomopt


mol = gto.Mole()
mol.atom="N; N 1, 1.1"
mol.basis= "6-31g"
mol.build()

mf = scf.RHF(mol).run()

mc = mcscf.CASCI(mf, 4,4)
mc.fcisolver = dmrgscf.DMRGCI(mc.mol)
mc.state_average_([0.25, 0.25, 0.25, 0.25])
excited_grad = mc.nuc_grad_method().as_scanner(state=2)
mol1 = geomopt.optimize(excited_grad)

Geometry optimization cycle 1
Cartesian coordinates (Angstrom)
 Atom        New coordinates             dX        dY        dZ
   N   0.000000   0.000000   0.000000    0.000000  0.000000  0.000000
   N   1.100000   0.000000   0.000000    0.000000  0.000000  0.000000
converged SCF energy = -108.867618373058
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/geomopt/__init__.py", line 21, in optimize
    return berny_solver.optimize(method, *args, **kwargs)
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/geomopt/berny_solver.py", line 164, in optimize
    **kwargs)[1]
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/geomopt/berny_solver.py", line 131, in kernel
    energy, gradients = g_scanner(mol)
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/grad/casci.py", line 223, in __call__
    e_tot = mc_scanner(mol)
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/mcscf/casci.py", line 543, in __call__
    e_tot = self.kernel(mo_coeff, ci0)[0]
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/mcscf/casci.py", line 827, in kernel
    self._finalize()
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/mcscf/addons.py", line 680, in _finalize
    old_finalize()
  File "/home/chenh/anaconda3/lib/python3.6/site-packages/pyscf/mcscf/casci.py", line 838, in _finalize
    for i, e in enumerate(self.e_cas):
TypeError: 'NoneType' object is not iterable
```

**Comment by sunqm** [comment]
This bug is fixed in PR #305. Besides the fix in PR #305, there are problems in state-average mcscf gradients code which were fixed in PR #339. In the next major release (pyscf-1.7, currently the dev branch), you should be able to optimize molecular geometry with the state-averaged DMRGSCF method.

Relevant examples can be found in https://github.com/pyscf/pyscf/blob/dev/examples/geomopt/12-mcscf_excited_states.py

**Comment by jamcas** [comment]
@sunqm, that's great !
